src/signTx.c

```diff
@@ -362,6 +384,14 @@ static void signTx_handleInit_ui_runStep()
 		);
 	}
 
+	UI_STEP(HANDLE_INIT_STEP_SCRIPT_RUNNING_WARNING) {
```

Comment:
I think we want to make this more general --- there might be other types of warnings in the future. The security policy resulting in the warning depends on tx init APDU only, so we can determine the reason also here, and then choose appropriate warning msg.

I don't think we should use UI_STEP_JUMP at all here. Because if security policy determined a warning should be displayed, then we must display it, or assert failure and stop processing because we have some bug.

For now, I'd be OK with just asserting that `numCollaterals > 0` and that tx signing mode is the new one for Plutus/Alonzo; we can extend it easily in the future if there are more possible warnings.

Reply:
Good that you brought this up here, because there WILL be more kinds of warnings, and I was already thinking how to incorporate it into the UI machine without making it overly complex or verbose. Is the UI step really the good place to make complicated warnings?
I used the jump because I assumed we can warn for a variety of reasons, since the warning step was already there and printing its text out in case of network id or protocol issues. Do we want a separate warning for those as well? This would basically copy the code of the security policy to the UI.

Reply:
For user experience, it would be nice if we could pack the warning into a single screen. Because usually you just click through the warning when you know you want to run scripts. Then UI_STEP would be OK. 

For network id / protocol magic, I think we are displaying them in HANDLE_INIT_STEP_DISPLAY_DETAILS, do we display any warning? Maybe Trezor does. Please consult with Gabo / David, if we can make it more consistent, then we should try.

For more complex warnings, I'm not sure how to approach that, but so far I don't know of any use case for anything more than a single screen.